### PR TITLE
Update tests to use networking.k8s.io/v1

### PR DESCRIPTION
**Pull request by lucacome**
I had to use k8s client version coming from this brach https://github.com/kubernetes-client/python/commits/release-19.0 that has the new v1 API

### CODE REVIEW

File: perf-tests/data/appprotect-ingress.yaml [hunk 1]
```diff
@@ -19,9 +19,13 @@ spec:
       paths:
       - path: /backend1/v1
         backend:
```
Comment (pleshakov) [inline perf-tests/data/appprotect-ingress.yaml:21]: pathType?

File: perf-tests/data/non-ap-ingress.yaml [hunk 2]
```diff
@@ -13,9 +13,13 @@ spec:
       paths:
       - path: /backend1
         backend:
-          serviceName: backend1-svc
-          servicePort: 80
+          service:
```
Comment (pleshakov) [inline perf-tests/data/non-ap-ingress.yaml:16]: pathType?

File: tests/Makefile [hunk 3]
```diff
@@ -29,9 +29,10 @@ run-tests-in-kind:
 
 .PHONY: create-kind-cluster
 create-kind-cluster:
-	kind create cluster
+	$(eval K8S_VERSION=$(shell grep "K8S_VERSION:" ../.github/workflows/edge.yml | awk -F" " '{print $$2}'))
```
Comment (pleshakov) [inline tests/Makefile:32]: nice

File: tests/data/appprotect/grpc/ingress.yaml [hunk 4]
```diff
@@ -21,5 +21,7 @@ spec:
       paths:
       - path: /helloworld.Greeter
```
Comment (pleshakov) [inline tests/data/appprotect/grpc/ingress.yaml:22]: pathType?

File: tests/docker/Dockerfile [hunk 5]
```diff
@@ -1,16 +1,16 @@
...
-COPY tests/requirements.txt /workspace/tests/ 
-RUN pip install -r requirements.txt 
+COPY tests/requirements.txt /workspace/tests/
+RUN pip install -r requirements.txt
 
 COPY tests /workspace/tests
 COPY deployments /workspace/deployments
 
-ENTRYPOINT ["python3", "-m", "pytest"]
+ENTRYPOINT ["python3", "-m", "pytest", "suite/"]
```
Comment (pleshakov) [inline tests/docker/Dockerfile:16]: why this change?

Comment (lucacome) [reply tests/docker/Dockerfile:16]: installing `kubernetes` from github will pull the `src` folder and pytest will run the tests for the library too

File: perf-tests/data/appprotect-ingress.yaml (hunk 1 above)
Comment (lucacome) [reply perf-tests/data/appprotect-ingress.yaml:22]: added

File: tests/data/appprotect/grpc/ingress.yaml (hunk 4 above)
Comment (lucacome) [reply tests/data/appprotect/grpc/ingress.yaml:22]: added

File: perf-tests/data/non-ap-ingress.yaml (hunk 2 above)
Comment (lucacome) [reply perf-tests/data/non-ap-ingress.yaml:17]: added

**Review by pleshakov — APPROVED** [review]

**Review by soneillf5 — APPROVED** [review]

**Review by vepatel — APPROVED** [review]
